components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.uma.resource.endpoint.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.ResourceEndpointConstants;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.ResourceRegistrationApiService;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.CreateResourceDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.ErrorDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.ReadResourceDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.ResourceDetailsDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.exceptions.ResourceEndpointException;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.util.ResourceUtils;
+
+import org.wso2.carbon.identity.oauth.uma.resource.service.ResourceConstants;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAClientException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAServiceException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.model.Resource;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.ws.rs.core.Response;
+
+/**
+ * ResourceRegistrationApiServiceImpl is used to handling resource management.
+ */
+public class ResourceRegistrationApiServiceImpl extends ResourceRegistrationApiService {
+
+    private static final Log log = LogFactory.getLog(ResourceRegistrationApiServiceImpl.class);
+
+    /**
+     * Register a resource with resource details
+     *
+     * @param requestedResource details of the resource to be registered
+     * @return Response with the status of the registration
+     */
+    @Override
+    public Response registerResource(ResourceDetailsDTO requestedResource) {
+
+        if (requestedResource == null) {
+            log.error("Request body cannot be empty.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        try {
+            Resource registerResource = ResourceUtils.getResourceService()
+                    .registerResource(ResourceUtils.getResource(requestedResource));
+            CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
+            return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+        } catch (UMAServiceException e) {
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
```

Comment:
Are we doing a special thing in this debug log? seems to be we log the same in error handling.